TrafficSimulator/Assets/RoadGenerator/Core/Runtime/Objects/Intersection.cs

```diff
@@ -14,12 +14,20 @@ public enum IntersectionType
         FourWayIntersection
     }
 
+    public enum FlowType
+    {
+        TrafficLights,
+        StopSigns,
+    }
+
     [RequireComponent(typeof(MeshFilter), typeof(MeshRenderer))]
     [ExecuteInEditMode()]
     [Serializable]
     public class Intersection : MonoBehaviour
     {
         [HideInInspector] public GameObject IntersectionObject;
+        [HideInInspector] public GameObject FlowContainer;
+        [HideInInspector] public GameObject TrafficLightController;
```

Comment:
Both these should be private. The trafficLightController gameobject should also be named with object in the end to make it clearer that it is a game object rather than a script